packages/hoppscotch-common/src/helpers/RequestRunner.ts

```diff
@@ -175,15 +186,36 @@ export function runRESTRequest$(
       requestHeaders = [...tab.value.document.request.headers]
     }
 
+    const finalRequestVariables =
+      tab.value.document.request.requestVariables.map(
+        (v: HoppRESTRequestVariable) => {
+          if (v.active) {
+            return {
+              key: v.key,
+              value: v.value,
+              secret: false,
+            }
+          }
+          return []
```

Comment:
```suggestion
          return {}
```
Is it possible to filter out the inactive entries instead of preserving empty entries? Is it intentional to keep the list length the same as the original one?